Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -2950,7 +2974,7 @@ template <class TreeBuilder> TreeClassExpression Parser<LexerType>::parseClass(T
     JSTextPosition classHeadEnd = lastTokenEndPosition();
     consumeOrFail(OPENBRACE, "Expected opening '{' at the start of a class body");
 
-    AutoPopScopeRef classScope(this, pushScope());
+    AutoPopScopeRef classScope(this, pushScope(setIsStaticBlockBoundary));
```

Comment:
Why not calling `setStaticBlock` after `pushScope()` instead of passing this parameter? The other scope creation is not doing this.